### ISSUE TITLE
Disabled attribute seems to have no effect.

### ISSUE THREAD

**Issue by germandiagogomez** [mikke89/RmlUi#461]
Hello.

It could be my rookie fault but I cannot make disabled attribute to work. I tried to set directly to try:

```
<button data-attrif-disabled="condition"/>
```

And directly to give it a check:

```
<button disabled/>
```

Clicks still work, the button has normal color and on hover it shows the hover color. Am I doing anything wrong?

**Comment by mikke89** [comment]
Hey there.

First of all, `<button>` has no particular meaning in RML, and thus, the disabled attribute does not have any effect on it. What you perhaps want is to use `<input type="button">`, which [does support this attribute](https://mikke89.github.io/RmlUiDoc/pages/rml/forms.html#form-children).

Secondly, there are no built-in styles in RmlUi, so you need to apply the desired style to it yourself. This also applies to the disabled state of form elements. For this, you can use the `:disabled` pseudo class selector, or attribute selector, to style its color and other visual aspects.

**Comment by germandiagogomez** [comment]
Thanks for the ultra-fast reply. I did not know that. In fact button renders stuff and RmlDebugger does not emit errors. :) That made me confused.

I just have one more thing left that I am not sure why it is not working for me. Some of my `input.button`s have a decorator. When I put the :disabled property on, it works, but still changes color on hover and they do not grey out correctly.

For buttons without gradient decorators it does not happen.

**Comment by mikke89** [comment]
Yeah, it's not an error to use that name. In fact, you can use any name for your tags in RML, only some types have a special meaning as listed in the documentation.

What is important to understand here is that, the disabled state doesn't stop the element from receiving hover and active states. So it will still match these rules from your style sheet. To make it act as you intend, you need to ensure that the properties from those rules are overridden in the disabled rule. Thus, you need to ensure that the *specificity* of the various rules are as intended so that the disabled rule takes precedence. This generally [works just like in CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity).

**Comment by germandiagogomez** [comment]
Thanks for your help. I will close this issue.
